components/playlist/browser/media_detector_component_manager.cc

```diff
@@ -103,14 +106,27 @@ bool MediaDetectorComponentManager::ShouldHideMediaSrcAPI(
                               });
 }
 
-const std::string& MediaDetectorComponentManager::GetMediaDetectorScript() {
-  if (!script_.empty())
-    return script_;
+std::string MediaDetectorComponentManager::GetMediaDetectorScript(
+    const GURL& url) {
+  std::string detector_script = script_;
+  if (detector_script.empty()) {
+    // In case we have yet to fetch the script, use local script instead. At the
+    // same time, fetch the script from component.
```

Comment:
is the plan to have the script eventually loaded from component updater? or will it continue to be hosted in this repository?

Reply:
> is the plan to have the script eventually loaded from component updater? or will it continue to be hosted in this repository?

We're loading the script from the component updater. `RegisterIfNeeded()` will trigger it. But we're using script embedded in this repository when it's test or the component isn't updated yet.

The current workflow is quite clumsy.
1. Write and test js code in this repository
2. Go through Code review process
3. When it's merged, dump it to Playlist component repository.
4. When it's dumped, update required c++ changes here again. - in this PR case, combining site specific script from the component updater would be this part.

So, we have a plan to change the workflow like so..
1. Write and test code in **playlist component repository**
2. Go through Code review process there
3. When it's merged, update DEPS so that `gclient sync` can update the local script directory.

Reply:
thank you for clarifying! i would say then it's pretty important to make sure that component has branch protection and code review requirements set up.

Reply:
Yep absolutely. I can confirm that :) thanks!